packages/edit-post/src/components/header/mode-switcher/index.js

```diff
@@ -46,13 +47,15 @@ function ModeSwitcher() {
 				select( editorStore ).getEditorSettings().codeEditingEnabled,
 			isEditingTemplate:
 				select( editorStore ).getRenderingMode() === 'template-only',
+			isEditingPattern:
+				select( editorStore ).getRenderingMode() === 'pattern-only',
 			mode: select( editPostStore ).getEditorMode(),
 		} ),
 		[]
 	);
 	const { switchEditorMode } = useDispatch( editPostStore );
 
-	if ( isEditingTemplate ) {
+	if ( isEditingTemplate || isEditingPattern ) {
```

Comment:
I think it's ok to allow the pattern to be edited in text mode, so can probably drop this addition.